### PR TITLE
[DBClusterParameterGroup] Handle Create Call while Stack Having IAM Role with Tag Condition

### PR DESCRIPTION
**Pull request by moataz-mhmd**
*Description of changes:*
When customer specify stack IAM role with tag conditions. Create call will fail with `AccessDenied` exception because it contains only system tags and apply other tags by `AddTagsToResource`.

I.e: Customer created this stack with stack level tag `foo -> bar`
```
AWSTemplateFormatVersion: 2010-09-09
Description: RDS DBClusterParameterGroup Example Stack

Resources:
  RDSDBClusterParameterGroup:
    Type: AWS::RDS::DBClusterParameterGroup
    DeletionPolicy: Delete
    Properties:
      Parameters:
        character_set_database: utf32
      Family: aurora5.6
      Description: "Any Description"
```

And assigned IAM Role to execute this stack
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": "rds:*",
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "rds:req-tag/foo": "bar"
                }
            }
        }
    ]
}
```
That means `CreateDBClusterParameterGroup` call need to have `--tags foo -> bar` Or the whole call will fail with `AccessDenied` Exception. 

**Approach**
Change create to include all possible tags (Stack, System, and Resource) together.
If failed with access denied that means there is possibility to apply soft fail for stack level tags, We trigger create call again with system tags only to ensure they are attached as they do not need permission. Then code flow will follow regular soft fail procedure for stack level tags.
 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java [hunk 1]
```diff
@@ -35,18 +36,19 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(final Amaz
...
 
         final Tagging.TagSet extraTags = Tagging.TagSet.builder()
                 .stackTags(Tagging.translateTagsToSdk(request.getDesiredResourceTags()))
                 .resourceTags(Translator.translateTagsToSdk(request.getDesiredResourceState().getTags()))
                 .build();
 
+        final Tagging.TagSet allTags = extraTags.toBuilder()
+                .systemTags(Tagging.translateTagsToSdk(request.getSystemTags()))
+                .build();
+
```
Comment (osdrv) [inline aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:48]: nit: I would suggest flipping it the other way around: all tags is the primary declaration and the extra tags as a subset.

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java [hunk 2]
```diff
@@ -35,18 +36,19 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(final Amaz
...
         return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
                 .then(progress -> setDbClusterParameterGroupNameIfMissing(request, progress))
-                .then(progress -> createDbClusterParameterGroup(proxy, proxyClient, progress, systemTags))
+                .then(progress -> createDbClusterPGWithAllTagsOrSystemTags(proxy, proxyClient, progress, allTags))
```
Comment (osdrv) [inline aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:51]: nit: too much implementation details in the method name. Consider: `safeCreateDBClusterParameterGroup`

**Review by Outige — APPROVED** [review]
LGTM

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java [hunk 3]
```diff
@@ -73,6 +75,18 @@ private ProgressEvent<ResourceModel, CallbackContext> createDbClusterParameterGr
                 });
     }
 
+    private ProgressEvent<ResourceModel, CallbackContext> createDbClusterPGWithAllTagsOrSystemTags(final AmazonWebServicesClientProxy proxy,
+                                                                                        final ProxyClient<RdsClient> proxyClient,
+                                                                                        final ProgressEvent<ResourceModel, CallbackContext> progress,
+                                                                                        final Tagging.TagSet allTags) {
+        ProgressEvent<ResourceModel, CallbackContext> progressEvent = createDbClusterParameterGroup(proxy, proxyClient, progress, allTags);
+        if(HandlerErrorCode.AccessDenied.equals(progressEvent.getErrorCode())) { //Resource is subject to soft fail on stack level tags.
```
Comment (osdrv) [inline aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:83]: nit: code formatting

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java [hunk 4]
```diff
@@ -73,6 +75,18 @@ private ProgressEvent<ResourceModel, CallbackContext> createDbClusterParameterGr
...
+            Tagging.TagSet systemTags = Tagging.TagSet.builder().systemTags(allTags.getSystemTags()).build();
+            return createDbClusterParameterGroup(proxy, proxyClient, progress, systemTags);
```
Comment (osdrv) [inline aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:85]: nit: Here is an idea: `createDbClusterParameterGroup` might set a flag like: `taggingOk: true`, which would be checked by `updateTags` and skip this step if it was an optimistic tagging case.

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java (hunk 3 above)
Comment (moataz-mhmd) [reply aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:83]: done!

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java (hunk 2 above)
Comment (moataz-mhmd) [reply aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:51]: Done! I updated method logic and naming. Please check if you have better naming.

File: aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java (hunk 1 above)
Comment (moataz-mhmd) [reply aws-rds-dbclusterparametergroup/src/main/java/software/amazon/rds/dbclusterparametergroup/CreateHandler.java:48]: Done!

**Review by osdrv — APPROVED** [review]

**Review by Outige — APPROVED** [review]

**Pull request merged**